ShotgunImportExport/Sources/CSVImportExport.cs

```diff
@@ -0,0 +1,123 @@
+        timeline.name = k_timelineObjectName;
+        director.playableAsset = timeline;
+        
+        foreach (var row in ReadCsvFile(Path.Combine(k_pathToFile, k_fileName)))
+        {
+            var CsvField = CreateActivationTrackAndClip(timeline, director);
+            CsvField.Item1.name = row["Shot Code"];
+            CsvField.Item2.start =  (Convert.ToDouble(row["Cut In"]) / k_FPS);
+            CsvField.Item2.duration = ((Convert.ToDouble(row["Cut Out"]) - Convert.ToDouble(row["Cut In"])) / k_FPS);
+        }
+    }
+
+    [UnityEditor.MenuItem("Shotgun CSV/Export")]
+    public static void ExportCsv()
+    {
+        GameObject director = GameObject.Find(k_directorObjectName);
+
+        // Get the actual timeline object
+        var timelineAsset = (director.GetComponent("PlayableDirector") as PlayableDirector).playableAsset as TimelineAsset;
+
+        // Each dictionary in the list represents a row in the CSV file
+        // Each value in the row is addressable by its column name
+        var toWrite = new List<Dictionary<string, string>>();
+
+        foreach (var track in timelineAsset.GetOutputTracks())
+        {
+            var dict = new Dictionary<string, string>();
+            var start = Convert.ToInt64(track.start * k_FPS);
+            var end = Convert.ToInt64((track.duration + track.start) * k_FPS);
+            var shotCode = track.name;
+
+            dict.Add("Shot Code", shotCode);
+            dict.Add("Cut In", start.ToString());
+            dict.Add("Cut Out", end.ToString());
+            toWrite.Add(dict);
+        }
+
+        var pathToOutputFile = Path.Combine(k_pathToFile, "export_from_unity.csv");
+        Debug.Log(pathToOutputFile);
+        WriteCsv(toWrite, pathToOutputFile);
+    }
+
+    private static List<Dictionary<string, string>> ReadCsvFile(string filePath)
+    {
+        var output = new List<Dictionary<string, string>>();
+
+        using (var reader = new StreamReader(filePath))
+        {
+            var fieldNamesList = reader.ReadLine().Split(',');
+            // Trim the double quotes
+            fieldNamesList = fieldNamesList.Select(x => x.Replace("\"", "")).ToArray();
+            
+            string line = reader.ReadLine();
+
+            while (null != line)
+            {
+                var dict = new Dictionary<string, string>();
+
+                var values = line.Split(',');
+                // Trim the double quotes
+                values = values.Select(x => x.Replace("\"", "")).ToArray();
+
+                for (Int64 idx = 0; idx < fieldNamesList.Length; idx++)
+                {
+                    dict.Add(fieldNamesList[idx], values[idx]);
+                }
+
+                output.Add(dict);
+
+                line = reader.ReadLine();
+            }
+        }
+        return output;
+    }
+
+    private static void WriteCsv (List<Dictionary<string, string>> input, string filePath)
+    {
+        //Keys should be uniform

```

Comment:
Enforce the assumption: either by passing in the header row explicitly (or as the first row in a list of rows), or by checking that indeed the dictionaries are all the same. Otherwise you export broken data.

Reply:
meh; ignore that for sample code